include/linux/logbuff.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * (C) Copyright 2007
+ * Wolfgang Denk, DENX Software Engineering, wd@denx.de.
+ *
+ * (C) Copyright 2014
+ * Freescale Inc.
+ *
+ * (C) Copyright 2016
+ * Mentor Graphics, Inc.
+ *
+ * See file CREDITS for list of people who contributed to this
+ * project.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of
+ * the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ * MA 02111-1307 USA
+ */
+#ifndef _LOGBUFF_H_
+#define _LOGBUFF_H_
+
+#ifdef CONFIG_LOGBUFFER
+extern void setup_ext_logbuff(void);
+#else
+static void inline setup_ext_logbuff(void) { }
+#endif /* CONFIG_LOGBUFFER */
+
+#endif /* _LOGBUFF_H_ */
```

Comment:
Actually, there is a printk.h, but in the process of discovering that, I found that there is already a setup_log_buf() being called in main.c, so I dropped this file entirely and added the call to setu_ext_logbuff() inside that function, with compile guards.